client/my-sites/themes/thanks-modal.jsx

```diff
@@ -201,20 +201,22 @@ const ThanksModal = React.createClass( {
 
 export default connect(
 	( state, { site } ) => {
+		let siteIdOrWpcom;
 		if ( ! site ) {
-			return {};
+			siteIdOrWpcom = 'wpcom';
+		} else {
+			siteIdOrWpcom = isJetpackSite( state, site.ID ) ? site.ID : 'wpcom';
 		}
 
-		const siteIdOrWpcom = isJetpackSite( state, site.ID ) ? site.ID : 'wpcom';
-		const currentThemeId = getActiveTheme( state, site.ID );
+		const currentThemeId = getActiveTheme( state, site && site.ID );
```

Comment:
If we put the `site &&` guard into the argument, we rely on the selector to do the right thing for a `false`y value. I think it'd be safer to put it before the selector, i.e.

```js
const currentThemeId = site && getActiveTheme( state, site.ID );
```

where expression evaluation short-circuiting will make sure `getActiveTheme()` won't even be called if the expression before the `&&` is falsey. (Same below)